server/v2/cometbft/abci.go

```diff
@@ -145,6 +155,38 @@ func (c *Consensus[T]) CheckTx(ctx context.Context, req *abciproto.CheckTxReques
 	return cometResp, nil
 }
 
+// checkTxFeeWithMinGasPrices ensure that the provided fees meet a minimum threshold for the validator,
+// if this is a CheckTx. This is only for local mempool purposes, and thus
+// is only ran on check tx.
+func (c *Consensus[T]) checkTxFeeWithMinGasPrices(tx transaction.Tx) error {
```

Comment:
how does this get overridden? previously it was an optional antehandler, now it seems mandatory. Although it could be set to 0, i believe we are pushing implementation details of the state machine into consensus here. Is there no way to get this in the antehandler like before?

Reply:
I tried to pass the config to TxValidator but couldn't find the best way to do it. Instead, I added it directly in CheckTx, tested it, and everything worked fine.

Reply:
Even if we add a check in `TxValidator`, it will still be executed in `stf.validateTx`, which is also called from `CheckTx`.

Reply:
The issue is getting the config value at the right place. Before it was in baseapp thus present on the SDK context.
If we put it in a TxValidator, we need to add that config in x/auth constructor (if we put it in x/auth) -- probably a viper like x/upgrade. Which makes it weird. We could abstract it as we already supply a viper, and have a provider directly in runtime/v2 (to keep viper out of auth).
But then there's the question which server component will hold that field.

If it is in comet, that config handling is super simple, however I do agree that it feels as well weird (both do tbh).